docs/developer-guide/api-overview.md

```diff
@@ -20,153 +20,154 @@ The relevant modules are available in Maven Central:
 interface compatible as per
 [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle) 
-and therefore they form two separate class hierarchies. The nature of the content,
-streaming vs full, is reflected in the type signatures of the content accessors:
+and therefore they form two separate class hierarchies. This is evident in the
+HTTP content accessors:
 
-* Streaming HTTP messages: 
+* `LiveHttpRequest` (and `LiveHttpResponse`): 
 
 ```java
-    public class HttpRequest { 
+    public class LiveHttpRequest { 
         ...
         
-        public StyxObservable<ByteBuf> body() { .. }
+        public ByteStream body() { .. }
     }
 ```
 
-* Full HTTP messages:
+* `HttpRequest` (and `HttpResponse`)
    
 ```java
-    public class FullHttpRequest { 
+    public class HttpRequest { 
         ...       
         public byte[] body() { .. }
         
         public String bodyAs(Charset charset) { .. }        
     }
 ```
 
-### Conversion Between HTTP Message Types
- 
-It is easy to convert between the streaming and full HTTP messages.  
+### Conversion Between HTTP Message Types 
 
-Streaming `HttpRequest` has a `toFullRequest` method (`toFullResponse` for `HttpResponse`).
-It aggregates a HTTP message body stream into one continuous byte array and creates a 
-corresponding `FullHttpRequest` (`FullHttpResponse`). Its type signature is:
+Call the `aggregate` method to convert a `LiveHttpRequest` (`LiveHttpResponse`) 
+to an immutable `HttpRequest` (`HttpResponse`) object: 
 
 ```java
-   public StyxObservable<FullHttpRequest> toFullRequest(int maxContentBytes);
+   public Eventual<HttpRequest> aggregate(int maxContentBytes);
 ```
 
-Note that the HTTP content streams represents live network traffic streaming
-through Styx core. The content stream is not memoized nor stored. 
-When the content is gone, it is gone for good. 
-Therefore a `HttpRequest` (`HttpResponse`) can be aggregated to a full 
-message strictly once only.
+This aggregates a HTTP message body stream and creates a corresponding immutable 
+HTTP object. Note that `aggregate` consumes the underlying live HTTP message `ByteStream`.
 
 The type signature illustrates other facts about the content stream:
 
-* Return type of `StyxObservable` shows that HTTP content stream is 
-  aggregated asynchronously. Obviously because the rest of the content 
-  stream are yet to be received from the network.
+* Return type of `Eventual` shows that `aggregate` is asynchronous. 
+  Obviously because it needs to wait for all of the byte stream to be fully 
+  available.
 
-* The `maxContentBytes` sets an upper limit for the message 
-  body, protecting Styx from exhausting memory in face of very long message
-  bodies, or perhaps from denial of service attacks. If the limit is reached,
-  the conversion fails with `ContentOverflowException`. 
+* The `maxContentBytes` sets an upper limit for the aggregated message 
```

Comment:
Perhaps rewrite as:

The `maxContentBytes` sets an upper limit for the aggregated message 
  body. This acts as a safety valve, protecting Styx from exhausting the heap memory when there are very long message bodies. 
  If the limit is reached, the conversion fails with `ContentOverflowException`. 

(I left out the part about DoS attacks since it works purely on the total body size and is not affected by the number of requests).

Reply:
ok